staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_whitebox_test.go

```diff
@@ -221,7 +221,14 @@ func TestGetListCacheBypass(t *testing.T) {
 	})
 	t.Run("ConsistentListFromCache", func(t *testing.T) {
 		featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.ConsistentListFromCache, true)
+		// TODO: fix me the following tests assume that etcdfeature.DefaultFeatureSupportChecker.Supports(storage.RequestWatchProgress)
```

Comment:
We should address this.

I'm not sure I'm following this comment - are you saying that now we have a race that the feature is not necessarily initialized correctly when the test is running?

Reply:
yes, I think so. The race exists today since this PR didn't introduce the test. This PR only adds a new case.

I have just run the `TestGetListCacheBypass` against the master branch and it fails:
```
=== RUN   TestGetListCacheBypass/ConsistentListFromCache
    cacher_whitebox_test.go:273: Unexpected bypass result for List request with options {ResourceVersion: ResourceVersionMatch: Predicate:{Label:<nil> Field:<nil> GetAttrs:<nil> IndexLabels:[] IndexFields:[] Limit:0 Continue: AllowWatchBookmarks:false} Recursive:false ProgressNotify:false SendInitialEvents:<nil>}, bypass expected: false, got: true
    --- FAIL: TestGetListCacheBypass/ConsistentListFromCache (0.00s)
 ```

Reply:
tmp fix in https://github.com/kubernetes/kubernetes/pull/125551